src/iosource/IOSource.h

```diff
@@ -7,7 +7,6 @@ extern "C" {
 }
 
 #include <string>
-#include "FD_Set.h"
 #include "Timer.h"
 
 namespace iosource {
```

Comment:
As the changes to the IOSource API are very small (which is nice!), I'm wondering if existing plugins may continue to compile? If so, can we add some piece that will actively break that so that people realize their plugins need updating?

Reply:
Existing plugins will fail for a couple of reasons:
1) methods that are declared with `override` that no longer exist in IOSource will cause a compiler error
2) new methods in IOSource are marked pure virtual and won't exist, and the compiler will error

Reply:
Slight modification to the above, since (1) is the only one that's actually the case. The `GetFds` and `NextTimestamp` methods were removed from IOSource, so anything using them and declaring them with `override` will fail to build.

Reply:
...  meaning if those methods don't have `override`, the plugin will continue to compile? Then I'm thinking it'd actually be worth adding some artificial breaking change that forces compilation to abort.

Reply:
I marked `GetNextTimeout` as pure virtual. That will cause old plugin code to fail to build. Sources that don't care about it can just return -1.

Out of curiosity, are there a lot of plugins out there that aren't packet sources? Packet source plugins would be masked from any changes in `IOSource` by `PktSrc` since it implements everything for them.

Reply:
Honest answer: I don't know. Probably not *that* many, but we may save somebody time/frustration by making clear what's happening.